tests/spec/core/beforeSave-spec.js

```diff
@@ -0,0 +1,55 @@
+import {
+  flushIframes,
+  getExportedDoc,
+  makeRSDoc,
+  makeStandardOps,
+} from "../SpecHelper.js";
+
+describe("Core - beforeSave config option", () => {
+  afterAll(flushIframes);
+
+  it("allows modification before saving", async () => {
+    const ops = makeStandardOps();
+    ops.config = null; // use src doc's config
+    const doc = await makeRSDoc(ops, "spec/core/beforeSave-spec.html");
+    expect(doc.getElementById("p1")).toBeNull();
+    expect(doc.getElementById("p2")).toBeNull();
+    const exportedDoc = await getExportedDoc(doc);
+    expect(exportedDoc.querySelectorAll("#p1").length).toBe(1);
+    expect(exportedDoc.querySelectorAll("#p2").length).toBe(1);
+    // make sure that the functions are run in order...
+    const p1 = exportedDoc.querySelector("#p1");
+    const p2 = exportedDoc.querySelector("#p2");
+    expect(p1.nextElementSibling).toBe(p2);
+  });
+
+  it("complains if it's not passed an array", async () => {
+    const ops = makeStandardOps({ beforeSave: "not a array" });
+    const doc = await makeRSDoc(ops);
+    const errors = doc.respec.errors.filter(
+      err => err.plugin === "core/beforeSave"
+    );
```

Comment:
Nit: can move this repeated code block into a function `getErrors(doc)` to the parent `describe` block.

Reply:
Added UIMessageFilters, and errorFilters, and warningFilters exports... wdyt?